src/boot/linux.c

```diff
@@ -56,22 +53,12 @@ static EFI_STATUS load_via_boot_services(
                 uint32_t compat_entry_point,
                 const char16_t *cmdline,
                 const struct iovec *kernel,
-                const struct iovec *initrd) {
+                const struct iovec *initrd,
+                EFI_DEVICE_PATH *kernel_file_path) {
```

Comment:
I'd like to make the function take `KERNEL_FILE_PATH *` rather than `EFI_DEVICE_PATH *`.